### PR TITLE
[CELEBORN-474][FOLLOWUP] Use JavaUtils#newConcurrentHashMap to speed up ConcurrentHashMap#computeIfAbsent

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

Use `JavaUtils#newConcurrentHashMap` to speed up `ConcurrentHashMap#computeIfAbsent`.

Follow up #1383.

### Why are the changes needed?

Celeborn supports JDK8, which could meet the bug mentioned in [JDK-8161372](https://bugs.openjdk.org/browse/JDK-8161372). Therefore, it's better to use `JavaUtils#newConcurrentHashMap` to speed up `ConcurrentHashMap#computeIfAbsent`.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

CI.

### CODE REVIEW

**Comment by SteNicholas** [comment]
Ping @turboFei.

File: openapi/openapi-client/src/main/java/org/apache/celeborn/rest/v1/master/invoker/ApiClient.java [hunk 1]
```diff
@@ -108,8 +109,8 @@ public class ApiClient extends JavaTimeFormatter {
 
   private Map<String, Authentication> authentications;
 
-  private Map<Long, Integer> lastStatusCodeByThread = new ConcurrentHashMap<>();
-  private Map<Long, Map<String, List<String>>> lastResponseHeadersByThread = new ConcurrentHashMap<>();
+  private Map<Long, Integer> lastStatusCodeByThread = JavaUtils.newConcurrentHashMap();
+  private Map<Long, Map<String, List<String>>> lastResponseHeadersByThread = JavaUtils.newConcurrentHashMap();
```
Comment (turboFei) [inline openapi/openapi-client/src/main/java/org/apache/celeborn/rest/v1/master/invoker/ApiClient.java:113]: These files are generated by openapi-generator,
we should not change it.

Comment (turboFei) [reply openapi/openapi-client/src/main/java/org/apache/celeborn/rest/v1/master/invoker/ApiClient.java:113]: And it is only used for client end, it should be fine.

**Review by turboFei — APPROVED** [review]
LGTM

**Comment by SteNicholas** [comment]
Ping @FMX, @waitinfuture.

**Review by FMX — APPROVED** [review]
LGTM. Thanks. Merged into main(v0.6.0).

**Pull request closed without merging**